src/main/java/io/curity/identityserver/plugin/events/listeners/config/AWSEventListenerConfiguration.java

```diff
@@ -42,6 +42,9 @@
     @Description("The AWS Region where DynamoDB is deployed. Use standard AWS region format, ex. us-east-2.")
     String getAwsRegion();
 
+    @Description("Enable this if the EC2 instance that the Curity Identity Server is running on has been assigned an IAM Role with permissions to DynamoDB. If this is enabled no Access Key ID, Access Key Secret, Aws Profile Name or Aws Role Arn are needed.")
+    Optional<Boolean> isUseEC2InstanceProfile();
```

Comment:
It sounds like this belongs in a choice. To do this, use `OneOf`. Here's some examples:

* [CriiptoAuthenticatorPluginConfig](https://github.com/curityio/criipto-authenticator/blob/77bbeaa2a024ddc5138ce6e4f51728150510c6ef/src/main/java/io/curity/identityserver/plugin/criipto/config/CriiptoAuthenticatorPluginConfig.java#L51)
* [AttributesConfiguration](https://github.com/curityio/rest-dap-plugin/blob/42c0e0d6cc37c1d02af5ae14115b33a018d71a73/src/main/java/io/curity/identityserver/plugin/data/access/json/config/AttributesConfiguration.java)
* [SignicatAuthenticatorPluginConfig](https://github.com/curityio/signicat-authenticator/blob/435b53da1398dc7ce46f87b45b73eed3cdd3b280/src/main/kotlin/io/curity/identityserver/plugin/signicat/config/SignicatAuthenticatorPluginConfig.kt#L52)
* [SlackAuthenticatorPluginConfig](https://github.com/curityio/slack-authenticator/blob/aa2e46f44b325d4c09ef6d0dcefc53620986e90d/src/main/java/io/curity/identityserver/plugin/slack/config/SlackAuthenticatorPluginConfig.java#L50)

Using this, you can make either this setting or all three others (access key ID, key secret and AWS profile name) required but never both.

Reply:
Added a choice option for the 3 possible different configurations used for DynamoDB access.